### PR TITLE
Translate snmp field values

### LINKED ISSUE #8458: Enable snmptranslate to field values
## Feature Request

The current implementation for the snmp input plugin does not enable snmptranslate for the field values, so there are some cases when the user is left with OID values for fields. I ran into this when ENTITY-MIB::entPhysicalVendorType returned an OID value. 

### Proposal:
Enable translating value for individual fields

### Current behavior:
E.g.
```
[[inputs.snmp.table]]
    oid = "ENTITY-MIB::entPhysicalTable"
 
    [[inputs.snmp.table.field]]
      oid = "ENTITY-MIB::entPhysicalVendorType"
```

### Desired behavior:
```
[[inputs.snmp.table]]
    oid = "ENTITY-MIB::entPhysicalTable"
 
    [[inputs.snmp.table.field]]
      oid = "ENTITY-MIB::entPhysicalVendorType"
      translate = true
```

### Use case: 
I have the need this kind of behavior enabled and all the prerequisites for snmptranslate already exist. I'd be happy to implement this if other people feel that this feature is good enough to be included in the snmp plugin.

### PR DESCRIPTION
**Pull request by LeadingMoominExpert**
### Required for all PRs:

- [X] Signed [CLA](https://influxdata.com/community/cla/).
- [X] Associated README.md updated.
- [X] Has appropriate unit tests.

Fixes #8458

### CODE REVIEW

File: plugins/inputs/snmp/snmp.go [hunk 1]
```diff
@@ -460,6 +462,21 @@ func (t Table) Build(gs snmpConnection, walk bool) (*RTable, error) {
...
+					case string:
+						{
+							entOid := ent.Value.(string)
+							_, _, oidText, _, err := SnmpTranslate(entOid)
+							if err == nil {
+								// If no error translating, the original value for ent.Value should be replaced
+								ent.Value = oidText
+							}
+						}
+					}
```
Comment (srebhan) [inline plugins/inputs/snmp/snmp.go:477]: I think this is overly complex. How about
```suggestion
					if entOid, ok := ent.Value.(string); ok {
						_, _, oidText, _, err := SnmpTranslate(entOid)
						if err == nil {
							// If no error translating, the original value for ent.Value should be replaced
							ent.Value = oidText
						}
					}
```

**Review by srebhan — COMMENTED** [review]
@LeadingMoominExpert you need to add the description of the new option also to the `sampleConfig` variable.
Furthermore, I'm not sure if your testcase actually is suited to test the translate functionality as you specify the name in the `tbl`. How can you see if it is *actually* overwritten using the translation?

File: plugins/inputs/snmp/snmp.go (hunk 1 above)
Comment (LeadingMoominExpert) [reply plugins/inputs/snmp/snmp.go:477]: Your'e right, the switch is redundant here.

**Comment by LeadingMoominExpert** [comment]
@srebhan I modified the tests so that a field returns an OID value with cases of this value being translated and not translated with mocked snmptranslate command.

**Comment by LeadingMoominExpert** [comment]
I also did not include the translate functionality in `sampleConfig` as it seems to only have descriptions of the plugin definitions and not table or field definitions. 
https://github.com/influxdata/telegraf/blob/a51a9a86a7b9f1be510b6c1a9e8ef9ef17aad9a2/plugins/inputs/snmp/snmp.go#L25-L69

**Review by srebhan — APPROVED** [review]
Fair enough. :-) Looks good.

**Review by ssoroka — APPROVED** [review]

**Comment by Hipska** [comment]
Does this also solve #6330? And could you give an example output with the function enabled? Thanks.

**Comment by ssoroka** [comment]
Looks like it doesn't solve it for integer fields.

**Pull request merged**